packages/block-template-react/src/app.tsx

```diff
@@ -2,41 +2,101 @@ import {
+ */
 import styles from "./base.module.scss";
 
+/**
+ * This defines the properties of the entity your block expects to be sent.
+ * This entity is available to your block on props.graph.blockEntity.
+ * To change the structure of the entity your block expects, change this type.
+ */
 type BlockEntityProperties = {
   name: string;
 };
 
+/**
+ * This function is to help illustrate a property being change when the button is pressed.
+ * You can remove it when you remove the button below.
+ */
+const supplyRandomName = () => {
+  const names = ["Alice", "Bob", "Carol", "Dave", "Erin", "Frank"];
+  return names[Math.floor(Math.random() * names.length)];
+};
+
+/**
+ * This is the entry point for your block – the function that embedding applications will call to create it.
+ * It is a function that takes a property object (known as "props" in React) and returns an element.
+ * You should update this comment to describe what your block does, or remove the comment.
+ */
 export const App: BlockComponent<BlockEntityProperties> = ({
   graph: {
+    /**
+     * The properties sent to the block represent the messages sent automatically from the application to the block.
+     * All block <> application messages are split into services, and so is this property object.
+     * Here, we're extracting the 'graph' service messages from the property object.
+     * – and then taking a single message from it, 'blockEntity'
+     * @see https://blockprotocol.org/docs/spec/graph-service#message-definitions for other such messages
+     */
     blockEntity: { entityId, properties },
   },
 }) => {
+  /**
+   * These two lines establish communication with the embedding application.
+   * You don't need to change or understand them, but if you are curious:
+   * 1. we create a 'ref' which will store a reference to an element – we need an element to communicate to the app via
+   *   - the ref stores 'null' at first, but will be attached to the root element in our block when it exists
+   * 2. we then feed the reference to a 'hook' (a function that uses React features), which sets up the graph service:
+   *   - this takes care of the lower-level details of communicating with the embedding application
+   *   - it returns a 'graphService' which has various methods on it, corresponding to messages your block can send
+   *   - see an example below for sending an 'updateEntity' message, and a link to the other available messages
+   */
   const blockRootRef = useRef<HTMLDivElement>(null);
   const { graphService } = useGraphBlockService(blockRootRef);
 
-  const updateSelf = useCallback(
-    (newProperties: Partial<BlockEntityProperties>) =>
-      graphService?.updateEntity({
-        data: { properties: newProperties, entityId },
-      }),
-    [entityId, graphService],
-  );
-
+  /** Here we extract the 'name' property from the blockEntity's properties ("object destructuring") */
```

Comment:
Hmm, not sure if we should explain ""object destructuring".

Reply:
Removed in https://github.com/blockprotocol/blockprotocol/pull/598/commits/3c41770f16d43573e718be49b1e379a9d3444454